Quarkus 1.7.x: Gradle Kotlin project compilation error

**Describe the bug**
After upgrading my Quarkus (gradle, Kotlin) project from 1.6.1.Final to 1.7.0.Final, I get a compilation error 

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':common:quarkusPrepareTests'.
> Cannot change dependencies of dependency configuration ':common:kotlinCompilerPluginClasspath' after it has been resolved.

```



**To Reproduce**
Steps to reproduce the behavior:
1. Download reproducer project and unzip it [2020-08-25_kotlin-compiling-issues.zip](https://github.com/quarkusio/quarkus/files/5129529/2020-08-25_kotlin-compiling-issues.zip)
2. Run `./gradlew check` in the project directory
3. The build fails with the error shown above
4. Change the Quarkus version in gradle.properties to 1.6.1.Final and it works


**Environment (please complete the following information):**
 - Quarkus 1.7.0
- Windows 10
- JDK 11

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java
@@ -16,7 +16,6 @@ import org.gradle.api.GradleException;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.Task;
-import org.gradle.api.UnknownTaskException;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.ConfigurationContainer;
 import org.gradle.api.artifacts.ProjectDependency;
@@ -137,13 +136,12 @@ public class QuarkusPlugin implements Plugin<Project> {
                 javaPlugin -> {
                     project.afterEvaluate(this::afterEvaluate);
                     JavaCompile compileJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_JAVA_TASK_NAME);
-                    compileJavaTask.mustRunAfter(quarkusGenerateCode);
+                    compileJavaTask.dependsOn(quarkusGenerateCode);
                     quarkusGenerateCode.setSourceRegistrar(compileJavaTask::source);
 
-                    try {
-                        // TODO: support kotlin
-                        tasks.getByName("compileTestKotlin");
-                    } catch (UnknownTaskException noKotlin) {
+                    Task compileKotlinTask = tasks.findByName("compileKotlin");
+                    // TODO: proper support of kotlin
+                    if (compileKotlinTask == null) {
                         JavaCompile compileTestJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_TEST_JAVA_TASK_NAME);
                         compileTestJavaTask.dependsOn(quarkusGenerateCodeTests);
                         quarkusGenerateCodeTests.setSourceRegistrar(compileTestJavaTask::source);
@@ -188,6 +186,10 @@ public class QuarkusPlugin implements Plugin<Project> {
                     tasks.withType(Test.class).forEach(configureTestTask);
                     tasks.withType(Test.class).whenTaskAdded(configureTestTask::accept);
                 });
+        project.getPlugins().withId("org.jetbrains.kotlin.jvm", plugin -> {
+            Task compileKotlinTask = tasks.findByName("compileKotlin");
+            compileKotlinTask.dependsOn(quarkusGenerateCode, quarkusGenerateCodeTests);
+        });
     }
 
     private Set<Path> getSourcesParents(SourceSet mainSourceSet) {

--- a/integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleKotlinProjectBuildTest.java
+++ b/integration-tests/gradle/src/test/java/io/quarkus/gradle/MultiModuleKotlinProjectBuildTest.java
@@ -1,5 +1,7 @@
 package io.quarkus.gradle;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
 import java.io.File;
 
 import org.junit.jupiter.api.Test;
@@ -8,9 +10,9 @@ public class MultiModuleKotlinProjectBuildTest extends QuarkusGradleWrapperTestB
 
     @Test
     public void testBasicMultiModuleBuild() throws Exception {
-
         final File projectDir = getProjectDir("multi-module-kotlin-project");
-
-        runGradleWrapper(projectDir, "clean", "build");
+        final BuildResult build = runGradleWrapper(projectDir, "clean", "build");
+        assertThat(build.getTasks().get(":quarkusGenerateCode")).isEqualTo(BuildResult.SUCCESS_OUTCOME);
+        assertThat(build.getTasks().get(":compileKotlin")).isEqualTo(BuildResult.SUCCESS_OUTCOME);
     }
 }

